src/elastio-snap.c

```diff
@@ -2047,7 +2301,7 @@ static int __cow_load_section(struct cow_manager *cm, unsigned long sect_idx){
 	ret = __cow_alloc_section(cm, sect_idx, 0);
 	if(ret) goto error;
 
-	ret = file_read(cm->filp, cm->sects[sect_idx].mappings, cm->sect_size*sect_idx*8 + COW_HEADER_SIZE, cm->sect_size*8);
+	ret = file_read_block(cm->dev, cm->sects[sect_idx].mappings, cm->sect_size*sect_idx*8 + COW_HEADER_SIZE, SECTORS_PER_BLOCK);
```

Comment:
This is just an old hardcode I guess: 8 means 'eight sectors per block'

Reply:
Then maybe better to add some define with a name for that to get rid of the "magic number".

Reply:
I prefer to keep it as it is, for now, to make sure we understand this correctly. Otherwise, we might find difficulties in understanding why we need that SECTORS_PER_BLOCK (in case we don't really need it).

Reply:
Ok. Then I'll keep it in mind and write about it again in the following PRs )